### ISSUE TITLE
Different names for semantically distinct enums that happen to have the same set of values

### ISSUE THREAD

**Issue by MariaMozgunova** [tfranzel/drf-spectacular#1019]
Hi,

Thanks for the project and your great effort!

In the [FAQ section](https://drf-spectacular.readthedocs.io/en/latest/faq.html#i-get-warnings-regarding-my-enum-or-my-enum-names-have-a-weird-suffix) of the drf-spectacular documentation you state the following:
```
If you have multiple semantically distinct enums that happen to have the same set of values, and you want different names for them, this mechanism won’t work.
```

I have two distinct enums and they happen to have the same set of values. I searched similar issues but I can not seem to find a solution now.

My initial warning was the following:
```
Warning: encountered multiple names for the same choice set (SomeEnum). This may be unwanted even though the generated schema is technically correct. Add an entry to ENUM_NAME_OVERRIDES to fix the naming.
```
I use the same enum in multiple serializers. So I went ahead and overridden `ENUM_NAME_OVERRIDES` as follows:
```
'ENUM_NAME_OVERRIDES': {'SomeEnum': '<path_to_enum>'}
```

But right after that, I faced another issue. After overriding `ENUM_NAME_OVERRIDES`, I now have two semantically different enums (`SomeEnum` and `AnotherEnum`) with the same value set having the same name (`SomeEnum`) in the generated schema.

Overriding `AnotherEnum` like this generates a warning as well.
```
'ENUM_NAME_OVERRIDES': {'SomeEnum': '<path_to_enum>', 'AnotherEnum': '<path_to_another_enum>'}
```
This is the warning generated:
```
Error: ENUM_NAME_OVERRIDES has duplication issues. Encountered multiple names for the same choice set. Enum naming might be unexpected.
```
And now in the schema, I only have `AnotherEnum` and no `SomeEnum`. This is not expected behavior.

tldr; I want `SomeEnum` and `AnotherEnum` to have different names in the generated schema and I also want to eliminate the first warning I mentioned in this issue (`encountered multiple names for the same choice set (SomeEnum)`).

How can I proceed?

**Comment by tfranzel** [comment]
Hi @MariaMozgunova. thanks!

So this issue is a conceptual one and there is currently no solution to it due to several reasons.

> If you have multiple semantically distinct enums that happen to have the same set of values, and you want different names for them, this mechanism won’t work.

This remains true, even if you prodded different warning variations. The fact that you got closest to your expected outcome initially was a lucky accident. 

You need to understand that we encounter a anonymous set of choices and then look for a proper name. Internally, `ENUM_NAME_OVERRIDES` is reversed like this: `{(1,2,3): "SomeEnum"}.  So it is impossible to have 2 names for the same set (except for a special conflict resolution case).

At the end of the day, this is due to the way DRF and Django are designed. Any name of choice classes is lost and the sets are by design anonymous. I tried fixing this in https://code.djangoproject.com/ticket/34388#ticket but the second part was not accepted and so the choices remain anonymous.
